app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+    # Enforces unique not null constraint on username and email
+    validates :username, presence: true,
+              uniqueness: { case_sensitive: false },
+              length: {minimum: 3, maximum: 25}
+
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```

Comment:
Is the `\A` and `\z` part of the regex? It seems like it is going to match the literal "A" and "z", which is probably not intended.

Reply:
Yep it's part of the regex, \A matches the start of the string, \z matches end of the string.